nautobot/extras/templatetags/plugins.py

```diff
@@ -56,10 +56,14 @@ def _get_registered_content(obj, method, template_context, return_html=True):
 
 
 @register.simple_tag(takes_context=True)
-def plugin_buttons(context, obj):
+def plugin_buttons(context, obj, view="detail"):
     """
     Render all buttons registered by plugins
     """
+    if view == "list":
+        # The object_list.html template passes content_type.model_class as obj
```

Comment:
Is this true both for old-style ObjectListViews and also NautobotUIViewSet?

Reply:
NM, I see you're doing this in the shared generic template not the view code.

Reply:
It is a bit irrelevant though since the TemplateExtension only applies to a single model class by design?

Reply:
Yes because this is driven by the template which is the same in both cases (assuming normal implementation, which is also assumed in all the other TemplateExtension use cases).

Reply:
Somewhat yes. I was mostly trying to not reinvent the wheel. There's a few other things we should add support for in the future for list views, in line with our other UX API improvements. So I think we will be able to refactor TemplateExtensions then.